Playbooks/AS-PagerDuty-Integration/azuredeploy.json

```diff
@@ -0,0 +1,392 @@
+                                                    ]
+                                                },
+                                                "type": "If"
+                                            }
+                                        },
+                                        "runAfter": {},
+                                        "type": "Foreach"
+                                    }
+                                }
+                            },
+                            "expression": {
+                                "and": [
+                                    {
+                                        "equals": [
+                                            "@length(triggerBody()?['object']?['properties']?['relatedEntities'])",
+                                            0
+                                        ]
+                                    }
+                                ]
+                            },
+                            "type": "If"
+                        },
+                        "Create_PagerDuty_Event": {
+                            "runAfter": {
+                                "Condition_-_Check_Event_Severity": [
+                                    "Succeeded"
+                                ]
+                            },
+                            "type": "Http",
+                            "inputs": {
+                                "body": {
+                                    "client": "Microsoft Sentinel",
+                                    "client_url": "@triggerBody()?['object']?['properties']?['incidentUrl']",
+                                    "event_action": "trigger",
+                                    "payload": {
+                                        "custom_details": {
+                                            "entities": "@variables('Entities')",
+                                            "severity": "@triggerBody()?['object']?['properties']?['severity']",
+                                            "status": "@triggerBody()?['object']?['properties']?['status']",
+                                            "title": "@triggerBody()?['object']?['properties']?['title']"
+                                        },
+                                        "description": "@triggerBody()?['object']?['properties']?['description']",
+                                        "entities": "entities list",
+                                        "severity": "@variables('EventSeverity')",
+                                        "source": "@triggerBody()?['workspaceInfo']?['ResourceGroupName']",
+                                        "summary": "@triggerBody()?['object']?['properties']?['title']",
+                                        "timestamp": "@triggerBody()?['object']?['properties']?['createdTimeUtc']"
+                                    },
+                                    "routing_key": "@variables('IntegrationKey')"

```

Comment:
Utilize value of **IntegrationKey** from the parameter itself. Saving this value in a variable will expose the text in clear text.

Reply:
Understood. This is resolved. I see why avoiding variables when unnecessary is preferable, but out of curiosity, would I have alternatively been able to keep the variable and change it's type from string to securestring?